mkdocs/serve.py

```diff
@@ -71,7 +71,7 @@ def builder():
             theme=theme,
         )
         config['site_dir'] = tempdir
-        build(config, live_server=True)
+        build(config, live_server=True, clean_site_dir=True)
```

Comment:
1. Start `mkdocs serve`
2. view a markdown file in the browser
3. delete/rename that markdown file, refresh.

It wont be removed until `mkdocs serve` is restarted. We might be building in a temp directory, but we build in it multiple times.